File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT2_JAVA/src/Organisms/Animals/Human.java
package Organisms.Animals;

import Game.World;
import Organisms.Organism;

import java.awt.*;
import java.io.Serializable;

public class Human extends Animal implements Serializable {
    public boolean isSkillActive;
    public int roundsLeft;
    public int cooldown;
    public Human(World world, int x, int y) {
        super("YOU", "Human", new Color(226, 36, 255), 5, 4, 1, world,x,y);
        isSkillActive=false;
        roundsLeft=0;
        cooldown=0;
    }
    @Override
    public Human getChild(int x, int y)
    {
        return new Human(world,x,y);
    }
    @Override
    public void Action()
    {
        if(!isSkillActive && cooldown>0)
        {
            cooldown--;
        }
        if(isSkillActive)
        {
            roundsLeft--;
        }
        int newX=this.getX()+ world.HumanX*getSpeed();
        int newY=this.getY()+ world.HumanY*getSpeed();
        if(world.IsPlaceTaken(newX,newY))
        {
            Organism victim = world.FindOrganismByPosition(newX,newY).get();
            victim.Collision(this);
        }
        else
        {
            world.addToLog(this + "skacze na pole " + newX + " " + newY);
            this.setXY(newX,newY);
        }
        world.HumanX=0;
        world.HumanY=0;


        if(isSkillActive && roundsLeft<3)
        {
            if(world.getRandomNumber(2)==0)
            {
                this.speed=2;
            }
            else
            {
                this.speed=1;
            }
        }
        if(isSkillActive && roundsLeft==0)
        {
            isSkillActive=false;
            this.speed=1;
        }
    }
}

File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT3_PYTHON/Organisms/Animals/human.py
import sys
import random

import pygame

from Organisms.Animals.animal import Animal


class Human(Animal):
    def __init__(self, position_x, position_y, world):
        super().__init__(position_x, position_y, "Human", 'H', 5, 4, world, 1, (255,20,147))
        self.isskillactive = False
        self.cooldown = 0
        self.duration = 0


    def getchild(self, newposition):
        return Human(newposition[0], newposition[1], self.world)

    def action(self):
        self.world.game.paintWorld()
        x=0
        y=0
        print("waiting for input" + self.tostring())
        running = True
        while running:
            for event in pygame.event.get():
                if event.type == pygame.QUIT:
                    pygame.quit()
                    sys.exit()
                if event.type == pygame.KEYDOWN:
                    if event.key == pygame.K_LEFT:
                        x = -1
                        if self.world.ispositiongood(self.position_x + x, self.position_y + y):
                            running = False
                        else:
                            x=0
                            y=0
                    if event.key == pygame.K_RIGHT:
                        x = 1
                        if self.world.ispositiongood(self.position_x + x, self.position_y + y):
                            running = False
                        else:
                            x = 0
                            y = 0
                    if event.key == pygame.K_UP:
                        y = -1
                        if self.world.ispositiongood(self.position_x + x, self.position_y + y):
                            running = False
                        else:
                            x = 0
                            y = 0
                    if event.key == pygame.K_DOWN:
                        y = 1
                        if self.world.ispositiongood(self.position_x + x, self.position_y + y):
                            running = False
                        else:
                            x = 0
                            y = 0
                    if event.key == pygame.K_r:
                        if(self.cooldown == 0):
                            self.isskillactive = True
                            self.speed = 2
                            self.cooldown = 6
                            self.duration = 5
                            self.world.game.paintWorld()

        x = x * self.speed
        y = y * self.speed
        newx, newy = self.position_x+x, self.position_y+y

        if self.world.isplacetaken(newx, newy):
            ofiara = self.world.findorganismbyposition(newx, newy)
            ofiara.collision(self)
        else:
            self.world.logs.append(self.tostring() + " skacze na " + str(newx) + " " + str(newy))
            self.position_x = newx
            self.position_y = newy

        if(self.duration<=1):
            self.isskillactive = False
            if(self.cooldown>0):
                self.cooldown -= 1
        else:
            if(self.duration > 2):
                self.isskillactive = True
            else:
                i = random.randint(0,1)
                if(i==1):
                    self.speed = 2
                else:
                    self.speed = 1
            self.duration -= 1




